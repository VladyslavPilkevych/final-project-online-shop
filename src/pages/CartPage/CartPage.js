/* eslint-disable max-len */
import React, { memo } from 'react';
import { useNavigate } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';

import CartItem from '../../components/CartItem/CartItem';
import Advantages from '../../components/Advantages/Advantages';
import { ReactComponent as PayPallIcon } from '../../assets/icons/paypall.svg';
import numberWithSpaces from '../../utils/numberWithSpaces';

import { deleteAllCart } from '../../store/actionCreators/cartAC';

import styles from './CartPage.module.scss';

function CartPage() {
  const navigate = useNavigate();
  const dispatch = useDispatch();

  const dataCart = useSelector((state) => state.cart.dataCart) || [];

  const totalPrice = (parseInt(dataCart.map((item) => item.product.currentPrice * item.cartQuantity).reduce((acc, value) => acc + value, 0), 10));
  const orderPrice = numberWithSpaces(totalPrice + (totalPrice * 0.1) + 21);
  return (
    <section>
      <div className={styles.cartWrapper}>
        <h1 className={styles.cartPageTitle}>Shopping Cart</h1>
        <div className={styles.cartPageColumn}>
          <div className={styles.cartPageGridWrapper}>
            <div className={styles.cartItemHeaders}>
              <div className={styles.Item}>
                <p>Item</p>
              </div>
              <div className={styles.Price}>
                <p>Price</p>
              </div>
              <div className={styles.Qty}>
                <p>Qty</p>
              </div>
              <div className={styles.Subtotal}>
                <p>Subtotal</p>
              </div>
            </div>
            <CartItem />
            <div className={styles.CartButtonWrapper}>
              <button className={styles.CartButtonShopping} type="button" onClick={() => navigate({ pathname: '/' })}>
                Continue Shopping
              </button>
              <button className={styles.CartButtonClearCart} type="button" onClick={() => { dispatch(deleteAllCart()); window.scrollTo(0, 0); }}>
                Clear Shopping Cart
              </button>
            </div>
          </div>
          <div className={styles.summaryContainer}>
            <div className={styles.summaryWrapper}>
              <h2 className={styles.titleSummary}>Summary</h2>
              <div className={styles.summaryContent}>
                <div>
                  <p className={styles.summarytext}>Subtotal</p>
                </div>
                <div>
                  <p className={styles.summarytext}>
                    $
                    {numberWithSpaces(totalPrice)}
                  </p>
                </div>
              </div>
              <div className={styles.summaryContent}>
                <div>
                  <p className={styles.summarytext}>Shipping</p>
                </div>
                <div>
                  <p className={styles.summarytext}>$21</p>
                </div>
              </div>
              <div className={styles.summaryContent}>
                <div>
                  <p className={styles.summarytext}>Tax</p>
                </div>
                <div>
                  <p className={styles.summarytext}>10%</p>
                </div>
              </div>
              <div className={styles.summaryContent}>
                <div>
                  <p className={styles.summarytext}>Order Total</p>
                </div>
                <div>
                  <p className={styles.summarytext}>
                    $
                    {orderPrice}
                  </p>
                </div>
              </div>
              <div className={styles.summaryButtonWrapper}>
                <button className={styles.buttonProceed} type="button">
                  Proceed to Checkout
                </button>
                <button className={styles.buttonCheck} type="button">
                  Check out with
                  <PayPallIcon />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div className={styles.advantagesBackGround}>
        <Advantages />
      </div>
    </section>
  );
}

export default memo(CartPage);
