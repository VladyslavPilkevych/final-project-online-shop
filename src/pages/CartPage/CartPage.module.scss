@import '../../assets/styles/global.module.scss';

.cartWrapper {
  width: 100%;
  display: flex;
  margin: 0 auto;
  max-width: 1398px;
  padding-left: 10px;
  padding-right: 10px;
  box-sizing: border-box;
  flex-direction: column;
}
.cartPageTitle {
  color: $black;
  font-size: 32px;
  font-weight: 600;
  margin-top: 68px;
  line-height: 48px;
}
.cartItemHeaders {
  display: grid;
  font-size: 14px;
  font-weight: 600;
  margin-top: 29px;
  line-height: 21px;
  padding-bottom: 13px;
  justify-content: space-between;
  border-bottom: 1px solid $grey-star;
  grid-template-columns: 3fr 1fr 1fr 1fr 0.5fr;
}
.cartPageColumn {
  display: grid;
  column-gap: 20px;
  grid-template-columns: 2fr 1fr;
}

.CartButtonWrapper {
  margin-top: 25px;
}
.CartButtonShopping {
  font-size: 14px;
  cursor: pointer;
  font-weight: 600;
  line-height: 21px;
  padding: 8px 30px;
  margin-right: 7px;
  border-radius: 50px;
  box-sizing: border-box;
  color: $dark-grey-text;
  border: 2px solid $dark-grey-text;
}
.CartButtonClearCart {
  color: $white;
  font-size: 14px;
  cursor: pointer;
  font-weight: 600;
  padding: 8px 30px;
  line-height: 21px;
  background: $black;
  border-radius: 50px;
  border: 2px solid black;
}
.advantagesBackGround {
  background-color: $bg-white-color;
}
.summaryContainer {
  width: 100%;
  max-width: 430px;
  background-color: $bg-white-color;
}
.summaryWrapper {
  padding: 17px 30px;
}
.titleSummary {
  color: $black;
  font-size: 24px;
  font-weight: 600;
  line-height: 36px;
  margin-bottom: 19px;
  padding-bottom: 10px;
  border-bottom: 1px solid $grey-star;
}
.summaryContent {
  display: flex;
  margin: 15px 0;
  justify-content: space-between;
}
.summarytext {
  color: $black;
  font-size: 13px;
  font-weight: 600;
  line-height: 210%;
}
.summaryButtonWrapper {
  gap: 7px;
  display: flex;
  flex-direction: column;
}
.buttonProceed {
  border: none;
  color: $white;
  font-size: 14px;
  font-weight: 600;
  padding: 8px 0px;
  line-height: 21px;
  text-align: center;
  border-radius: 50px;
  background-color: $blue;
}
.buttonCheck {
  border: none;
  font-size: 14px;
  color: #232c65;
  font-weight: 600;
  padding: 8px 0px;
  line-height: 21px;
  text-align: center;
  border-radius: 50px;
  background-color: $orange;
}

@media (max-width: 768px) and (min-width: 500px) {
  .cartPageColumn {
    column-gap: 20px;
    grid-template-columns: 1fr 1fr;
  }
  .summaryWrapper {
    padding: 20px;
  }
  .cartPageTitle {
    font-size: 18px;
    line-height: 27px;
  }
  .cartItemHeaders {
    display: none;
  }
  .titleSummary {
    font-size: 14px;
    line-height: 21px;
  }
  .summarytext {
    font-size: 11px;
    line-height: 210%;
  }
  .buttonProceed {
    font-size: 13px;
    line-height: 20px;
  }
  .buttonCheck {
    font-size: 13px;
    line-height: 20px;
  }
  .CartButtonShopping {
    font-size: 13px;
    font-weight: 600;
    line-height: 20px;
    margin-bottom: 10px;
  }
  .CartButtonClearCart {
    font-size: 13px;
    font-weight: 600;
    line-height: 20px;
  }
  .CartButtonWrapper {
    display: flex;
    flex-direction: column;
  }
}
@media (max-width: 500px) and (min-width: 240px) {
  .cartPageColumn {
    column-gap: 20px;
    grid-template-columns: 1fr;
  }

  .summaryWrapper {
    padding: 20px;
  }
  .cartPageTitle {
    font-size: 18px;
    line-height: 27px;
  }
  .cartItemHeaders {
    display: none;
  }
  .titleSummary {
    font-size: 14px;
    line-height: 21px;
  }
  .summarytext {
    font-size: 11px;
    line-height: 210%;
  }
  .buttonProceed {
    font-size: 13px;
    line-height: 20px;
  }
  .buttonCheck {
    font-size: 13px;
    line-height: 20px;
  }
  .CartButtonShopping {
    margin-right: 0;
    font-size: 13px;
    font-weight: 600;
    line-height: 20px;
    margin-bottom: 10px;
  }
  .CartButtonClearCart {
    font-size: 13px;
    font-weight: 600;
    line-height: 20px;
  }
  .CartButtonWrapper {
    display: flex;
    flex-direction: column;
  }
  .cartPageGridWrapper {
    order: 1;
  }
}
