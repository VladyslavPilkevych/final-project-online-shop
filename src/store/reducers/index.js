import { combineReducers } from 'redux';
import preloaderReducer from './preloaderReducer';
import productsReducer from './productsReducer';
import menuReducer from './menuReducer';
import userReducer from './userReducer';
import searchReducer from './searchReducer';
import cartReducer from './cartReducer';
import miniMenuReducer from './miniMenuReducer';
import filtersCategoriesReducer from './filtersCategoriesReducer';

const reducer = combineReducers({
  loader: preloaderReducer,
  products: productsReducer,
  menu: menuReducer,
  miniMenu: miniMenuReducer,
  filtersCategories: filtersCategoriesReducer,
  user: userReducer,
  search: searchReducer,
  cart: cartReducer,
});

export default reducer;
