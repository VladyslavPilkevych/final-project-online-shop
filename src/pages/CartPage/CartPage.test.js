import { render } from '@testing-library/react';
import CartPage from './CartPage';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../store';

function Component() {
  return (
    <BrowserRouter>
      <Provider store={store}>
        <CartPage />
      </Provider>
    </BrowserRouter>
  );
}

describe('Snapshot CartPage should render', () => {
  test('Snapshot CartPage is rendered', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});