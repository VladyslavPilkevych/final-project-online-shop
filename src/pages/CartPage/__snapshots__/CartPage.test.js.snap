// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot CartPage should render Snapshot CartPage is rendered 1`] = `
<DocumentFragment>
  <section>
    <div
      class="cartWrapper"
    >
      <h1
        class="cartPageTitle"
      >
        Shopping Cart
      </h1>
      <div
        class="cartPageColumn"
      >
        <div
          class="cartPageGridWrapper"
        >
          <div
            class="cartItemHeaders"
          >
            <div
              class="Item"
            >
              <p>
                Item
              </p>
            </div>
            <div
              class="Price"
            >
              <p>
                Price
              </p>
            </div>
            <div
              class="Qty"
            >
              <p>
                Qty
              </p>
            </div>
            <div
              class="Subtotal"
            >
              <p>
                Subtotal
              </p>
            </div>
          </div>
          <div>
            <div
              class="cartItemContainer"
            >
              <ul />
            </div>
          </div>
          <div
            class="CartButtonWrapper"
          >
            <button
              class="CartButtonShopping"
              type="button"
            >
              Continue Shopping
            </button>
            <button
              class="CartButtonClearCart"
              type="button"
            >
              Clear Shopping Cart
            </button>
          </div>
        </div>
        <div
          class="summaryContainer"
        >
          <div
            class="summaryWrapper"
          >
            <h2
              class="titleSummary"
            >
              Summary
            </h2>
            <div
              class="summaryContent"
            >
              <div>
                <p
                  class="summarytext"
                >
                  Subtotal
                </p>
              </div>
              <div>
                <p
                  class="summarytext"
                >
                  $0.00
                </p>
              </div>
            </div>
            <div
              class="summaryContent"
            >
              <div>
                <p
                  class="summarytext"
                >
                  Shipping
                </p>
              </div>
              <div>
                <p
                  class="summarytext"
                >
                  $21
                </p>
              </div>
            </div>
            <div
              class="summaryContent"
            >
              <div>
                <p
                  class="summarytext"
                >
                  Tax
                </p>
              </div>
              <div>
                <p
                  class="summarytext"
                >
                  10%
                </p>
              </div>
            </div>
            <div
              class="summaryContent"
            >
              <div>
                <p
                  class="summarytext"
                >
                  Order Total
                </p>
              </div>
              <div>
                <p
                  class="summarytext"
                >
                  $21.00
                </p>
              </div>
            </div>
            <div
              class="summaryButtonWrapper"
            >
              <button
                class="buttonProceed"
                type="button"
              >
                Proceed to Checkout
              </button>
              <button
                class="buttonCheck"
                type="button"
              >
                Check out with
                <svg>
                  paypall.svg
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="advantagesBackGround"
    >
      <div
        class="advantagesWrapper"
      >
        <div
          class="advantagesCard"
        >
          <img
            alt="advantages-icon"
            class="advantagesIcon"
            src="account-icon.png"
          />
          <h3
            class="advantagesTitle"
          >
            Product Support
          </h3>
          <p
            class="advantagesText"
          >
            Up to 3 years on-site warranty available for your peace of mind.
          </p>
        </div>
        <div
          class="advantagesCard"
        >
          <img
            alt="advantages-icon"
            class="advantagesIcon"
            src="support-icon.png"
          />
          <h3
            class="advantagesTitle"
          >
            Personal Account
          </h3>
          <p
            class="advantagesText"
          >
            With big discounts, free delivery and a dedicated support specialist.
          </p>
        </div>
        <div
          class="advantagesCard"
        >
          <img
            alt="advantages-icon"
            class="advantagesIcon"
            src="saving-icon.png"
          />
          <h3
            class="advantagesTitle"
          >
            Amazing Savings
          </h3>
          <p
            class="advantagesText"
          >
            Up to 70% off new Products, you can be sure of the best price.
          </p>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
